src/poliastro/plotting/_base.py

```diff
@@ -159,6 +186,28 @@ def _plot(self, orbit, *, label=None, color=None, trail=False):
             coordinates, orbit.r, label=label, colors=colors, dashed=True
         )
 
+    def _anim(self, orbit, *, label=None, color=None, trail=False):
+        colors = self._get_colors(color, trail)
+
+        self.set_attractor(orbit.attractor)
+        from poliastro.twobody import propagation
+
+        orbit = orbit.change_plane(self.plane)
+
+        label = generate_label(orbit.epoch, label)
+        time_laps = [i for i in range(int(orbit.period.value))]
+        tofs = time_laps * u.day * 2
+        k = propagation.danby(orbit.attractor.k, orbit.r, orbit.v, tofs)[0]
```

Comment:
In order to properly visualize the change in velocity, an array of positions and associated times are required. This forces us to propagate the orbit, but, what about if the orbit is a perturbed one? Then, this would force us to make use of the cowell solver. Moreover, you are restricting time propagation to the period and hence, to elliptic orbits. What about open-shape ones like parabolic and hyperbolic?

Again, let me ping @astrojuanlu  for a better discussion on this.

Finally, the farnocchia solver has been proof to be accurate and robust. In fact, it is the default one in the `propagation` function. Is there any particular reason why you selected the danby one?

Reply:
yes, we can implement different solvers for different kind of orbits.
but there should be certain limits on animation like period and all because it is not good to have an animation that is running for a long or infinite time instead of this what we can do is we can give this control to the user for how long they want to see an animated plot for such orbits if no input is given we can put certain default value.
no there is no specific reason I opted for danby we can always change it with the best one as you said farnochia solver